Sources/Roboflow/Classes/RFClassificationModel.swift

```diff
@@ -0,0 +1,285 @@
+    public override func detect(pixelBuffer buffer: CVPixelBuffer, completion: @escaping (([RFPrediction]?, Error?) -> Void)) {
+        guard let mlModel = self.mlModel else {
+            completion(nil, NSError(domain: "RFClassificationModel", code: 1, userInfo: [NSLocalizedDescriptionKey: "Model initialization failed."]))
+            return
+        }
+        
+        do {
+            // Get the actual input name from model description
+            guard let inputName = mlModel.modelDescription.inputDescriptionsByName.keys.first,
+                let inputDescription = mlModel.modelDescription.inputDescriptionsByName[inputName] else {
+                completion(nil, NSError(domain: "RFClassificationModel", code: 2, userInfo: [NSLocalizedDescriptionKey: "Could not find model input name"]))
+                return
+            }
+            
+            
+            // Get expected image dimensions
+            let inputImage: MLFeatureValue
+            if let imageConstraint = inputDescription.imageConstraint {
+                let targetWidth = Int(imageConstraint.pixelsWide)
+                let targetHeight = Int(imageConstraint.pixelsHigh)
+                
+                // Create a resized pixel buffer
+                let resizedBuffer = resizePixelBuffer(buffer, targetWidth: targetWidth, targetHeight: targetHeight)
+                inputImage = MLFeatureValue(pixelBuffer: resizedBuffer ?? buffer)
+            } else {
+                // Use original buffer if no constraints
+                inputImage = MLFeatureValue(pixelBuffer: buffer)
+            }
+            
+            let inputDict = [inputName: inputImage]
+            let inputProvider = try MLDictionaryFeatureProvider(dictionary: inputDict)
+            
+            // Run prediction directly with CoreML
+            let prediction = try mlModel.prediction(from: inputProvider)
+            
+            // Get the actual output name from model description
+            guard let outputName = mlModel.modelDescription.outputDescriptionsByName.keys.first,
+                  let output = prediction.featureValue(for: outputName) else {
+                completion(nil, NSError(domain: "RFClassificationModel", code: 3, userInfo: [NSLocalizedDescriptionKey: "Could not find model output"]))
+                return
+            }
+            
+            var predictions: [RFPrediction] = []
+            
+            // Handle different output types
+            if let multiArray = output.multiArrayValue {
+                // Multi-array output (may be logits that need softmax conversion)
+                let rawValues = multiArray.dataPointer.bindMemory(to: Float.self, capacity: multiArray.count)
+                
+                // Check if values are logits (outside 0-1 range) and need softmax
+                var needsSoftmax = false
+                for i in 0..<multiArray.count {
+                    if rawValues[i] < 0.0 || rawValues[i] > 1.0 {
```

Comment:
this definitely could return a wrong value of if it needs softmax or not -- i should think this would just be set on a per architecture basis